src/lazy/expanded/e_expression.rs

```diff
@@ -56,35 +44,40 @@ impl<'top, D: Decoder> ArgGroup<'top, D> {
     pub fn raw_arg_group(&self) -> <D::EExp<'top> as RawEExpression<'top, D>>::ArgGroup {
         self.raw_arg_group
     }
-    pub fn invoked_macro(&self) -> MacroRef<'top> {
-        self.invoked_macro
-    }
-    pub fn expressions(&self) -> ArgGroupIterator<'top, D> {
-        ArgGroupIterator::new(self.context, self.raw_arg_group())
+    pub fn expressions(&self) -> EExpArgGroupIterator<'top, D> {
+        EExpArgGroupIterator::new(self.context, self.raw_arg_group())
     }
-    pub fn expand(&self, environment: Environment<'top, D>) -> IonResult<MacroExpansion<'top, D>> {
+    pub fn expand(&self) -> IonResult<MacroExpansion<'top, D>> {
```

Comment:
🗺️ Several types that previously required the `Environment` to be passed into their methods now capture the relevant `Environment` at construction time.